csrc/executor_utils.cpp

```diff
@@ -1260,11 +1259,106 @@ CompiledKernel getCompiledKernel(
       compiled_kernel.module,
       compiled_kernel.kernel_name.c_str()));
 
-  if (!return_compiled_binary) {
-    compiled_kernel.ptx.clear();
-    compiled_kernel.cubin.clear();
```

Comment:
I think previously when we call this function in `compileFusion`, we don't necessarily need to store the ptx or cubin unless we are dumping it earlier. Why do we need to change this behavior?

Reply:
If we don't save the cubin, then it won't be available for serialization later on.

The kernel database saves the cubin immediately in this function when it's enabled. It doesn't need to look up the cubin in CompiledKernel.

Reply:
Got'ya. Thanks for clarifying this.

Reply:
Why do we need to keep the compiled binary?

Reply:
We keep the compiled binary, so it can be saved during serialization and to avoid recompiling with NVRTC.

Reply:
Is serialization always enabled?

I wonder if the memory usage could become a concern.

Reply:
Users can call serialize and deserialize through the python frontend, so it is always available.
```python
fc = FusionCache.get()
fc.serialize(tmp.name)

FusionCache.reset()

# Get new FusionCache because the previous one was destroyed by the reset call.
fc = FusionCache.get()
fc.deserialize(tmp.name)
```

Storing ~100 cubins is about 1.3 MB. Majority of the cubins are less than 10 KB. The largest cubin is ~32 KB.
I think this is reasonable.

Reply:
I see. That makes sense. Thanks!